### DISCUSSION TITLE
how to get something that describes what needs to change?

### DISCUSSION THREAD

**Discussion by maxandersen** [pulumi/pulumi#15194] [General]
I have a import.json with resources I've scraped from github.

Now i run pulumi import and it picks them up and generate a .java file with the code for the imported config.

All good.

Now I run pulumi preview using a different github owner (to try and test without modifying the original org) and I would expect it says it needs to create the resources but instead it is stating I need to unprotect the resources.

I unprotect all of them and now when I run pulumi preview I'm told all the repos  (the resources) are to be deleted...which I find counter intuitive.

What am I not grokkin here - what are the steps to take to have pulumi state what it wants to create/modify ?

**Comment by sam-cogan** [comment]
Look at the full diff of the preview, it should show you what has changed that means wants to destroy the resources for recreation.

Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________
From: Max Rydahl Andersen ***@***.***>
Sent: Sunday, January 21, 2024 9:36:57 AM
To: pulumi/pulumi ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [pulumi/pulumi] how to get something that describes what needs to change? (Discussion #15194)


I have a import.json with resources I've scraped from github.

Now i run pulumi import and it picks them up and generate a .java file with the code for the imported config.

All good.

Now I run pulumi preview using a different github owner (to try and test without modifying the original org) and I would expect it says it needs to create the resources but instead it is stating I need to unprotect the resources.

I unprotect all of them and now when I run pulumi preview I'm told all the repos (the resources) are to be deleted...which I find counter intuitive.

What am I not grokkin here - what are the steps to take to have pulumi state what it wants to create/modify ?

**Reply by maxandersen** [reply]
Full log is where ? I just get list of resource deletes. 

Also maybe I'm doing something wrong or misunderstanding about pulimi. Ie. Pulumi should never delete a github repository and recreate it - that would be devastating as then you for example loose all your stars/follow on a repo. 

Hence why I'm trying to ensure it will do an update rather than a destroy/delete.

**Comment by sam-cogan** [comment]
When you do the Pulumi up command you should get a prompt asking you if you want to go ahead, with options for yes, no and details, select details and it will show the full diff.  You can also run Pulumi preview with the —diff flag.

As with Terraform and most other IaC languages, Pulumi will attempt to recreate a resource if you are trying to change a property that the resource does not allow you to change on an active resource.

The other reason why it may be trying to delete is because it thinks the resource  is no longer defined in your code. Make sure that names and identifiers match exactly what is in GitHub.


Sent from Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Reply by maxandersen** [reply]
They are made using import so should all match. 

I'll try create reproducer to share.
